Source/Core/ClippingPlaneCollection.js

```diff
@@ -93,56 +115,146 @@ define([
          * @type {Boolean}
          * @default true
          */
-        combineClippingRegions : {
+        unionClippingRegions : {
             get : function() {
-                return this._combineClippingRegions;
+                return this._unionClippingRegions;
             },
             set : function(value) {
-                if (this._combineClippingRegions !== value) {
-                    this._combineClippingRegions = value;
-                    this._testIntersection = getTestIntersectionFunction(value);
+                if (this._unionClippingRegions !== value) {
+                    this._unionClippingRegions = value;
+                    this._testIntersection = value ? unionIntersectFunction : defaultIntersectFunction;
                 }
             }
         }
     });
 
-    function getTestIntersectionFunction(combineClippingRegions) {
-        if (combineClippingRegions) {
-            return function(value) {
-                return (value === Intersect.INSIDE);
-            };
+    /**
+     *
+     * @param plane
+     *
+     * @exception {DeveloperError} The plane added exceeds the maximum number of supported clipping planes.
+     *
+     * @see ClippingPlaneCollection#remove
+     * @see ClippingPlaneCollection#removeAll
+     */
+    ClippingPlaneCollection.prototype.add = function(plane) {
+        this._planes.push(plane);
+
+        //>>includeStart('debug', pragmas.debug);
+        if (this.length > ClippingPlaneCollection.MAX_CLIPPING_PLANES) {
+            throw new DeveloperError('The maximum number of clipping planes supported is ' + ClippingPlaneCollection.MAX_CLIPPING_PLANES);
+        }
+        //>>includeEnd('debug');
+    };
+
+    /**
+     * Returns the plane in the collection at the specified index.  Indices are zero-based
+     * and increase as planes are added.  Removing a plane shifts all planes after
+     * it to the left, changing their indices.  This function is commonly used with
+     * {@link ClippingPlaneCollection#length} to iterate over all the planes
+     * in the collection.
+     *
+     * @param {Number} index The zero-based index of the plane.
+     * @returns {Plane} The plane at the specified index.
+     *
+     * @see ClippingPlaneCollection#length
+     */
+    ClippingPlaneCollection.prototype.get = function(index) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.typeOf.number('index', index);
+        //>>includeEnd('debug');
+
+        return this._planes[index];
+    };
+
+    function indexOf(planes, plane) {
+        for (var i = 0; i < planes.length; ++i) {
```

Comment:
Not sure if the performance matters anymore in practice, but the Cesium convention is to assign `planes.length` to a local instead of dereferencing each iteration of the loop.